projects/packages/forms/src/blocks/contact-form/components/step-controls/index.js

```diff
@@ -0,0 +1,261 @@
+			}
+
+			const selectedStepClientIdForForm = getActivePreviewStepId( formClientId );
+
+			return {
+				selectedStepId: selectedStepClientIdForForm,
+				isPreview: isPreviewMode( formClientId ),
+				containerId: containerClientId,
+			};
+		},
+		[ formClientId, clientId, steps ]
+	);
+
+	// Don't render controls if there are no steps
+	if ( ! steps || steps.length === 0 ) {
+		return null;
+	}
+
+	const { stepLabel, index: currentStepIndex, isFirstStep, isLastStep } = currentStepInfo;
+
+	// Format the display label
+	let displayLabel;
+	if ( ! isPreview ) {
+		displayLabel = __( 'All Steps', 'jetpack-forms' );
+	} else if ( currentStepIndex >= 0 ) {
+		displayLabel = `${ currentStepIndex + 1 }. ${ stepLabel }`;
+	} else {
+		displayLabel = __( 'Select Step', 'jetpack-forms' );
+	}
+
+	// Custom function to insert a step container block after a specific block
+	const insertStepAtIndex = ( targetId, index ) => {
+		// Create a new step container block with default attributes
+		const newStepBlock = createBlock(
+			'jetpack/form-step',
+			{
+				stepLabel: __( 'New Step', 'jetpack-forms' ),
+			},
+			[]
+		);
+
+		insertBlock( newStepBlock, index, targetId );
+
+		// Select the newly created block
+		selectBlock( newStepBlock.clientId );
+
+		// Set this as the preview step if in preview mode
+		if ( isPreview ) {
+			setPreviewStep( formClientId, newStepBlock.clientId );
+		}
+	};
+
+	return (
+		<BlockControls>
+			<ToolbarGroup>
+				{ showToggle && (
+					<>
+						<ToolbarButton
+							onClick={ () => disablePreview( formClientId ) }
+							isPressed={ ! isPreview }
+						>
+							{ __( 'All Steps', 'jetpack-forms' ) }
+						</ToolbarButton>
+						<ToolbarButton
+							onClick={ () => {
+								if ( ! isPreview && steps.length > 0 ) {
+									// First set the step if one isn't already selected
+									if ( selectedStepId === null ) {
+										setPreviewStep( formClientId, steps[ 0 ].clientId );
+									}
+									// Then enable preview mode
+									enablePreview( formClientId );
+								}
+							} }
+							isPressed={ isPreview }
+						>
+							{ __( 'Preview', 'jetpack-forms' ) }
+						</ToolbarButton>{ ' ' }
+					</>
+				) }
+
+				{ isPreview && showNavigation && (
+					<>
+						<ToolbarButton
+							showTooltip={ true }
+							label={ __( 'Previous Step', 'jetpack-forms' ) }
+							disabled={ isFirstStep }
+							onClick={ navigateToPreviousStep }
+						>
+							<Icon icon={ previous } />
+						</ToolbarButton>
+						<ToolbarButton
+							showTooltip={ true }
+							label={ __( 'Next Step', 'jetpack-forms' ) }
+							disabled={ isLastStep }
+							onClick={ navigateToNextStep }
+						>
+							<Icon icon={ next } />
+						</ToolbarButton>
+						<DropdownMenu
+							icon={ null }
+							label={ __( 'Select step to preview', 'jetpack-forms' ) }
+							popoverProps={ { placement: 'bottom-start' } }
+							toggleProps={ {
+								children: (
+									<>
+										{ displayLabel }
```

Comment:
Probably need some "Step" fallback here in case of empty string:

<img width="256" alt="image" src="https://github.com/user-attachments/assets/f7dcba06-e291-48c3-8bb0-8c1aae1aa8dd" />


Not super important for v1 but we might want to ensure very long step names don't break the toolbar:

<img width="731" alt="Screenshot 2025-05-13 at 16 20 49" src="https://github.com/user-attachments/assets/02169ef5-3e87-4009-9fec-9d9d9f24370b" />

We could just cut them with `…`:

<img width="628" alt="Screenshot 2025-05-13 at 16 21 11" src="https://github.com/user-attachments/assets/9fa6bfb8-0941-41e3-a004-63077068da05" />

Reply:
Done